packages/plugin-ext/src/main/browser/view/tree-views-main.tsx

```diff
@@ -218,9 +226,24 @@ export class TreeViewWidget extends TreeWidget {
         @inject(TreeProps) readonly treeProps: TreeProps,
         @inject(TreeModel) readonly model: TreeModel,
         @inject(ContextMenuRenderer) readonly contextMenuRenderer: ContextMenuRenderer,
-        @inject(TreeViewDataProviderMain) readonly dataProvider: TreeViewDataProviderMain) {
+        @inject(TreeViewDataProviderMain) readonly dataProvider: TreeViewDataProviderMain,
+        @inject(SelectionService) readonly selectionService: SelectionService) {
 
         super(treeProps, model, contextMenuRenderer);
+
+        this.toDispose.pushAll([
+            this.model.onSelectionChanged(selection => {
+                // Todo(vzhukovskyi) Uncomment this when https://github.com/theia-ide/theia/issues/4623 will be resolved
+                // if (this.applicationShell.activeWidget === this) {
+                this.selectionService.selection = selection;
+                // }
+            }),
+            Disposable.create(() => {
+                if (this.selectionService.selection === this) {
```

Comment:
just realized that this line bogus in the navigator, it should be `this.selectionService.selection === this.model.selectedNodes`

Reply:
Updated. Thanks!